### PR TITLE
Association setup

### PR DESCRIPTION
**Pull request by mercymugambi**
Project requirements
Create a class Classroom with the following:
@label instance variable, should be initialized in the constructor.
Setter and getter for @label (remember about attr_accessor).
Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
Create the has-many side (a classroom has many students).
Create the belongs-to side (a student belongs to a classroom).
Make sure that when adding a student to a classroom it also sets the classroom for the student.
Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
Create a class Book with the following:
@title and @author instance variables, should be initialized in the constructor.
Setters and getters for instance variables (remember about attr_accessor).
Create a class Rental with the following:
@date instance variable, should be initialized in the constructor.
Setter and getter for @date (remember about attr_accessor).
Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
Create the has-many side of Book and Rental (a book has many rentals).
Create the belongs-to side of Rental and Book (a rental belongs to a book).
Create the has-many side of Person and Rental (a person has many rentals).
Create the belongs-to side of Rental and Person (a rental belongs to a person).
Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require_relative 'person'
+
+class Rental
+  attr_accessor :person, :book, :date
+
+  def initialize(person, book, date)
+    @person = person
+    @book = book
+    @date = date
+  end
```
Comment (jerryowusu) [inline rental.rb]: The rental should add itself to the @book and @person to help achieve the `belongs-to` associations. Kindly consider doing something like below for both books and person to achieve that. :point_down:  (this is just a code sample)

             @book = book
             book.rentals << self

File: person.rb [hunk 2]
```diff
@@ -28,4 +31,9 @@ def can_use_services?
 
     false
   end
+
+  def add_rental(rental)
+    rental.person = self
+    @rentals << rental
+  end
```
Comment (jerryowusu) [inline person.rb]: In order to achieve the `has-many` associations, the person itself should be present as a parameter to create the new rental. Kindly consider creating a new rental in this method for adding rentals and pass the date and book  as parameters. It should be similar to something below: 

       def add_rental(book, date)
          Rental.new(date, book, self)
       end

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    rental.book = self
+    @rentals << rental
+  end
```
Comment (jerryowusu) [inline book.rb]: In order to achieve the `has-many` associations, the person itself should be present as a parameter to create the new rental. Kindly consider creating a new rental in this method for adding rentals and pass the date and person as parameters. It should be similar to something below: 

       def add_rental(person, date)
          Rental.new(date, self, person)
       end

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @mercymugambi 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

so-close


## Highlights
Set up associations :+1: 


## Required Changes ♻️

- [ ] In the student class, there should be a method that serves as the `belongs-to` side since the `student` belongs to the `classroom`. Kindly check [this link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships) for more examples on how to do that. 

_Check the comments under the review._

## Optional suggestions

- [x] There are a lot of bullet shapes in the readme as shown  below, kindly consider tidying up your file to make it look more proffessional and presentable :point_down: 

Screenshot from 2023-06-22 21-39-47


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mwanawabangona — APPROVED** [review]
Hi @mercymugambi 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**